### PR TITLE
fix: clamp fitness heatmap sticky-label segment to avoid a zero-width column

### PR DESCRIPTION
**Pull request by llun**
## Summary

Small follow-up to [#853](https://github.com/llun/activities.next/pull/853) (fitness heatmap / mobile box sizing). That PR was merged before this last edge-case fix from the review loop could land.

A final independent review pass found a reachable degenerate case in the calendar heatmap's sticky labels: after `pushLabel` nudges a colliding label to `weekIndex + 1`, a **minimum 7-day, week-aligned range that still crosses a month boundary** (e.g. `2024-01-29 → 2024-02-04`, where Jan 29 is a Monday) produces a **single grid week**. The nudged `Feb` label then computes a segment width of `(1 - 1) = 0`, so the label overflows the one-column grid.

Reachable only from the dashboard's exact minimum-range selection; the heatmap route's `getDateRange` (month/year-aligned, ~1y all-time) never hits it. No crash and no label overlap — purely a cosmetic overflow — but the `pushLabel` "the shifted column always exists" invariant was simply false for this case.

## Fix

- Clamp the trailing segment to at least one column in `StickyLabelRow` (`Math.max(1, …)`), so **no** input can produce a zero/negative-width segment regardless of range.
- Correct the now-inaccurate `pushLabel` comment.
- Add a regression test for the degenerate single-week, month-crossing range.

## Tests

`prettier --check`, `yarn lint` (0 errors), `yarn build`, and the heatmap test suite (6 tests) all pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/components/fitness/FitnessCalendarHeatmap.tsx [hunk 1]
```diff
@@ -232,8 +232,13 @@ const StickyLabelRow: FC<{
...
       {items.map((item, index) => {
         const next = items[index + 1]
-        const segmentWeeks =
+        // Clamp to at least one column: a label nudged off the end of a
+        // single-week grid (a minimum-range, week-aligned, period-crossing
+        // selection) would otherwise compute a zero-width segment.
+        const segmentWeeks = Math.max(
+          1,
           (next ? next.weekIndex : totalWeeks) - item.weekIndex
+        )
```
[the remark this answers is not in the document]
Comment (llun) [reply lib/components/fitness/FitnessCalendarHeatmap.tsx:241]: Done in `8c8a77c65` — switched to `next?.weekIndex ?? totalWeeks`. (`??` keeps a real `weekIndex` of 0 since it only falls back on null/undefined.)

**Comment by llun** [comment]
### Sub-agent review round (#854)

A reviewer agent confirmed the clamp is correct and behaviour-equivalent in the normal case, and that `next?.weekIndex ?? totalWeeks` preserves a real `weekIndex` of 0. It also pointed out the regression test was presence-only (a zero-width div still renders its text in jsdom, so it passed with or without the clamp). Strengthened in `6bd9d3993`: the test now asserts the nudged label's segment is not `0px` wide — verified it fails without the clamp and passes with it.
